### DISCUSSION TITLE
Issues with every step. Now I am stuck.

### DISCUSSION THREAD

**Discussion by YorkyPoo** [danielmiessler/Fabric#282] [Q&A]
I apologize for how bad I am... I am Muphy's Law w/ tech.
Each step of the installation gave me problems:
- I didn't know what terminal to use because every guide was using a Mac.
- I started on Powershell but my prompts weren't going through so I decided to use Command Prompt.
- Some of those were not going through so I searched and found out I needed to update my PowerShell.
- Then my Kali terminal was the only one working.
- Git was originally installed under User/name but in the end, it was saying it didn't see git but it knew it was there.
- GPT helped me with some of that but then it was saying it saw git but it didn't want to open it there.
- I then moved it all over to program files because the video I was watching had his there. 
- A little hick-up there just trying to find out how to get to Windows folders in WSL.
- Almost every step wasn't installing correctly and I was using GPT3.5 to walk me through new troubleshooting commands.
- Eventually, I did "fabric --help" and it showed me everything I wanted to see. Till,
- None of the promps were working. "-pt", "pbpaste". 
- Bash can't find anything
- There is no Docker. I installed that and ran it. Not sure what to do now.
- Commands can't find ChatGPT4. 
- --listmodels shows a bunch of models but I can't switch them.
-  "Fabric --gpt-3.5-turbo."  = "-bash: Fabric: command not found}
-  "--gpt-3.5-turbo" = "-bash: --gpt-3.5-turbo: command not found". I know there Miessler there is a server version you can set up if you want to but everything is native.

I've tried to update the Kali terminal as much as I could. I know it's not equipped with everything but It's the only one that gave me any progress. Ubunto and PowerShell aren't taking me anymore. If anyone can assist me that would be great. In the meantime, I am going to try to delete it all and just try to use Ubunto from the start and try to keep it at [~] so I don't have to cd every time I open it. I believe I'll have to move the git again but I'm not sure. Thank you.

**Comment by lcsmd** [comment]
I got a little disoriented when I had to use wsl to install pipx.  Turns out you can install pipx in windows using pip. 

First cd to the fabric folder 

     cd ../fabric


     pip install pipx

You can install ffmpeg using winget. 

     winget install ffmpeg

Then 

      pipx install .  

(After the word install type a space then a dot. )

Lastly run setup 

      fabric --setup

**Reply by YorkyPoo** [reply]
ok, The "pipx install  ." was screwing with me. Probably because it would not work and I didn't know if it was because the Kali terminal didn't like it, I didn't update that terminal correctly, or did I not install the right tools?

**Reply by lcsmd** [reply]
Try it in windows

**Reply by YorkyPoo** [reply]
So I started with PowerShell:
[C:\Users\slyre> cd c:\
c:\> cd "program files"
c:\Program Files>cd code
c:\Program Files\Code>cd Fabric]

[c:\Program Files\Code\fabric>pip install pipx
'pip' is not recognized as an internal or external command,
operable program or batch file.]

Same thing with Command Prompt

**Reply by YorkyPoo** [reply]
Trying to work through it with GPT3.5. pip and Python were not found in PowerShell. I downloaded and installed Python. Then I Identified its path in EVN under user/name and system. Restarted PowerShell and it still can't find Python or pip.

**Reply by YorkyPoo** [reply]
I think I am getting somewhere now. 
- I spent a couple of hours to find out I needed to install Python.
- A couple more hours to find out if I installed the recommended one online, Even if I updated the path it could not be found.
- I deleted that and installed the one from the Microsoft store.
- That ran well but pip and pipx Defaulting to user installation because normal site-packages is not writeable
Collecting pipx.
- Everything after that installed well but I had to copy all of their paths.
- I'm now stuck where I was stuck before. I can talk to the system with the PowerShell now but trying to do something I get error code 404 GPT-4 doesn't exist. I try to switch models by following the directions.
- According to the instructions, my default modle is GPT-4 but I don't have that. To change it I need to --clear. 
-  fabric --clear = usage: fabric [-h] [--text TEXT] [--copy] [--agents {tr....
- fabric --model = unrecognized arguments
-  fabric --model gpt-3.5-turbo = Enter Question:
- I ask it a philosophical question and It tells me I have insufficient funds.

**Comment by lcsmd** [comment]
I'm working on using the liteLLM 

https://github.com/BerriAI/litellm

It seems like it could work well with fabric and make it easy to try many llms and also track costs and maintain a persistent log

**Reply by YorkyPoo** [reply]
Someone else on here had a post about doing something on mobile too. I wanted to do that. I just need to find out if he was starting on the PC or if the prerequisites allowed him to put the commands in.

**Comment by YorkyPoo** [comment]
I had to ask GPT-3.5 what to do at every step, I had to use Python from the Microsoft store. Everything I installed I had to create a system pathway for.
